### PR TITLE
dhserv

### PR DESCRIPTION
**Pull request by minhtung0404**
This change is [Reviewable](https://reviewable.io/reviews/natsukagami/tet-school-2017/7)

### CODE REVIEW

File: 20170117/DHSERV.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+* Ban đầu các đỉnh đều bị tắt
+* Có 2 truy vấn:
+
+ 1.bật đỉnh thứ i. 
+
+ 2.tìm đường đi nhỏ nhất từ u đến v với u, v bất kì nhỏ hơn hoặc bằng n.
+
+## Required Skills
+* Thuật toán tìm đường đi ngắn nhất.
```
Comment (phucdtd) [inline 20170117/DHSERV.md]: Nói rõ thuật toán nào? (Floyd)

File: 20170117/DHSERV.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# DHSERV 
+## Nguyễn Minh Tùng
+
+## Problem Statement
+* Cho một đồ thị với n đỉnh và m cạnh
+
+* Chúng ta có thể đi từ đỉnh u đến v qua đỉnh s khi mà chỉ khi có cạnh nối giữa u-s và s-v và đỉnh s đã được bật
```
Comment (phucdtd) [inline 20170117/DHSERV.md]: Chúng ta có thể đi từ đỉnh u đến đỉnh v qua đỉnh s khi và chỉ khi có đường đi giữa u-s và s-v, và đỉnh s đã được bật.
Nói cách khác, 1 đường đi chỉ tồn tại khi và chỉ khi tất cả các đỉnh trên đường đi đó đã được bật

File: 20170117/DHSERV.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# DHSERV 
+## Nguyễn Minh Tùng
```
Comment (phucdtd) [inline 20170117/DHSERV.md]: ###[Problem Link](http://vn.spoj.com/problems/DHSERV/)

File: 20170117/DHSERV.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+* Cho một đồ thị với n đỉnh và m cạnh
+
+* Chúng ta có thể đi từ đỉnh u đến v qua đỉnh s khi mà chỉ khi có cạnh nối giữa u-s và s-v và đỉnh s đã được bật
+
+* Ban đầu các đỉnh đều bị tắt
+* Có 2 truy vấn:
+
+ 1.bật đỉnh thứ i. 
+
+ 2.tìm đường đi nhỏ nhất từ u đến v với u, v bất kì nhỏ hơn hoặc bằng n.
```
Comment (phucdtd) [inline 20170117/DHSERV.md]: Nếu không tồn tại đường đi in ra -1

File: 20170117/DHSERV.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+## Problem Statement
+* Cho một đồ thị với n đỉnh và m cạnh
+
+* Chúng ta có thể đi từ đỉnh u đến v qua đỉnh s khi mà chỉ khi có cạnh nối giữa u-s và s-v và đỉnh s đã được bật
+
+* Ban đầu các đỉnh đều bị tắt
+* Có 2 truy vấn:
+
+ 1.bật đỉnh thứ i.
```
Comment (phucdtd) [inline 20170117/DHSERV.md]: Truy vấn "1 i" : Bật đỉnh thứ i

File: 20170117/DHSERV.md (hunk 4 above)
Comment (phucdtd) [inline 20170117/DHSERV.md]: Truy vấn "2 u v": Tính độ dài đường đi ngắn nhất giữa 2 nút u và v (u,v &le; n)

**Comment by natsukagami** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7#-:-Kam-4VYmeOe92ncH1dN:bnfp4nl)*

**Comment by natsukagami** [comment]
Em chỉnh lại folder đến file là `20170117/Tuyen/DHSERV.md` nhé, sau đó là merge được rồi.

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7#-:-Kam-_zRt65wr0zbpvuY:bgfwfn9)*

**Review by phucdtd — CHANGES_REQUESTED** [review]
Có vẻ ổn cả rổi đấy

File: 20170117/DHSERV.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# DHSERV 
+* http://vn.spoj.com/problems/DHSERV/
```
Comment (phucdtd) [inline 20170117/DHSERV.md]: chỉnh 1 chút... chỗ này dùng markdown trông cho gọn
```Markdown
[Problem Link](http://vn.spoj.com/problems/DHSERV/)

File: 20170117/DHSERV.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+
+## Nguyễn Minh Tùng
+
+## Problem Statement
+* Cho một đồ thị với n đỉnh và m cạnh
+
+* Chúng ta có thể đi từ đỉnh u đến v qua đỉnh s khi mà chỉ khi có cạnh nối giữa u-s và s-v, và đỉnh s đã được bật. Hay nói một cách khác thì một đường đi giữa hai đỉnh tồn tại khi mà chỉ khi các đỉnh trung gian trên đường đi đã được bật.
```
Comment (phucdtd) [inline 20170117/DHSERV.md]: khi VÀ chỉ khi*

**Review by phucdtd — CHANGES_REQUESTED** [review]
Last review

**Comment by phucdtd** [comment]
*[20170117/DHSERV.md, line 2 at r2](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7#-KahU_L4JYD9DauodiLZ-r2-2:-Kam6AWIEbRiBVEA2EfX:bb6ywmx) ([raw file](https://github.com/natsukagami/tet-school-2017/blob/f2d29304ba4e71273083a327b9673cd346cf6d7b/20170117/DHSERV.md#L2)):*
<details><summary><i>Previously, phucdtd (Hoàng Bảo Phúc) wrote…</i></summary><blockquote>

chỉnh 1 chút... chỗ này dùng markdown trông cho gọn
```Markdown
[Problem Link](http://vn.spoj.com/problems/DHSERV/)
</blockquote></details>

LGTM

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7)*
<!-- Sent from Reviewable.io -->

**Comment by phucdtd** [comment]
*[20170117/DHSERV.md, line 10 at r2](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7#-KahU_L4JYD9DauodiLZ-r2-10:-Kam6GR9mpeBlMkYoCsl:bb6ywmx) ([raw file](https://github.com/natsukagami/tet-school-2017/blob/f2d29304ba4e71273083a327b9673cd346cf6d7b/20170117/DHSERV.md#L10)):*
<details><summary><i>Previously, phucdtd (Hoàng Bảo Phúc) wrote…</i></summary><blockquote>

khi VÀ chỉ khi*
</blockquote></details>

LGTM

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7)*

**Comment by natsukagami** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7)*

**Comment by natsukagami** [comment]
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7#-Kam7kZKVjAMipHZ4gkj:-Kam7kZKVjAMipHZ4gkk:b-844qzp) (no related file):*
Xóa file cũ đi em

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7)*

**Comment by natsukagami** [comment]
Reviewed 1 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7)*

**Comment by natsukagami** [comment]
Reviewed 1 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[a discussion](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7#-Kam7kZKVjAMipHZ4gkj:-KamDG_8-QTwaMGFenkZ:bnfp4nl) (no related file):*
<details><summary><i>Previously, natsukagami (Natsu Kagami) wrote…</i></summary><blockquote>

Xóa file cũ đi em
</blockquote></details>

:lgtm:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/natsukagami/tet-school-2017/7)*

**Pull request merged**